test/tasks/work_dir_test.rb

```
@@ -111,4 +112,42 @@ def test_work_dir_warns_when_directory_exists_without_marker
+        assert Dir.exist?(path)
+        # Should have marker file
+        assert File.exist?(File.join(path, Kompo::WorkDir::MARKER_FILE))
+      end
+
+      # Should warn about permission denied and cache invalidation
+      assert_match(/permission denied.*invalidating Ruby cache/i, err)
+      # Cache directory should be deleted because the old work_dir can't be recreated
+      # and the cached Ruby has paths hardcoded to the old work_dir
+      refute Dir.exist?(cache_dir), "Cache directory should be deleted when work_dir cannot be recreated"
+    end
+  end
```

RuboCop (1.82.1)

[convention] Assignment Branch Condition size for `test_work_dir_invalidates_cache_on_permission_denied` is too high. [<7, 22, 0> 23.09/23]

(Metrics/AbcSize)